src/character_functions.cpp

```diff
@@ -1177,11 +1177,16 @@ void find_ammo_helper( T &src, const item &obj, bool empty, Output out, bool nes
             if( node->is_magazine() ) {
 
                 if( !node->contents.empty() ) {
+                    bool match = false;
                     for( const ammotype &at : ammo ) {
-                        if( node->contents.front().ammo_type() != at ) {
-                            return VisitResponse::SKIP;
+                        if( node->contents.front().ammo_type() == at ) {
+                            match = true;
+                            break;
                         }
                     }
+                    if( !match ) {
+                        return VisitResponse::SKIP;
+                    }
```

Comment:
```suggestion
                    const bool match = std::ranges::any_of( ammo, [&]( const ammotype & at ) {
                        return node->contents.front().ammo_type() == at;
                    } );
                    if( !match ) return VisitResponse::SKIP;
```
maybe we could use [std::ranges::any_of](https://en.cppreference.com/w/cpp/algorithm/ranges/all_any_none_of) here

Reply:
Resylt should be the same, if a bit cleaner.
Tested on my end with both vanilla and modded again, still works as intended.